what4/src/What4/Protocol/VerilogWriter/AST.hs

```diff
@@ -275,6 +281,10 @@ data ModuleState sym n =
                 -- type, signedness, name, and initializer of each.
                 , vsFreshIdent :: Int
                 -- ^ A counter for generating fresh names.
+                , vsSeenNonces :: Map.Map Word64 Identifier
+                -- ^ A map from the identifier nonces seen so far to their identifier names.
+                , vsUsedIdentifiers :: Set.Set Identifier
+                -- ^ A set of all the identifiers used so far
```

Comment:
Is the set of values in `vsSeenNonces` equivalent to the `vsUsedIdentifiers`?  What's the relationship between these groups of identifiers?

Reply:
They are unfortunately not the same, due to `addFreshWire`. The changes in this PR complicate the logic somewhat with the goal that we can use original names as much as possible but still be guaranteed to have unique names everywhere.

Reply:
OK.  Maybe consider updating  the comments here to help describe the differences.